test/geo.test.js

```diff
@@ -131,3 +133,96 @@ describe('GeoPoint', function() {
     });
   });
 });
+
+describe('nearFilter', function() {
+  it('should handle an empty where gracefully', function() {
+    var result = nearFilter(null);
+    result.should.be.false;
+  });
+
+  it('should handle a simple where filter with location', function() {
+    var result = nearFilter({
+      location: {
+        near: { lat: 0, lng: 0 },
+      },
+    });
+
+    should.exist(result);
+    result.near.lat.should.equal(0);
+    result.near.lng.should.equal(0);
+    should.not.exist(result.maxDistance);
+    should.not.exist(result.unit);
+    result.key.length.should.be.exactly(1);
+    result.key[0].should.be.exactly('location');
+  });
+
+  it('should handle a simple where filter with location and maxDistance', function() {
+    var result = nearFilter({
+      location: {
+        near: { lat: 0, lng: 0 },
+        maxDistance: 100,
+      },
+    });
+
+    should.exist(result);
+    result.near.lat.should.equal(0);
+    result.near.lng.should.equal(0);
+    should.exist(result.maxDistance);
+    result.maxDistance.should.equal(100);
+    should.not.exist(result.unit);
+    result.key.length.should.be.exactly(1);
+    result.key[0].should.be.exactly('location');
+  });
+
+  it('should handle logical operators with location', function() {
+    var result = nearFilter({
+      and: [
+        {
+          location: {
+            near: { lat: 0, lng: 0 },
+          },
+        },
+        {
+          title: 'test',
+        },
+      ],
+    });
+
+    should.exist(result);
+    result.near.lat.should.equal(0);
+    result.near.lng.should.equal(0);
+    should.not.exist(result.maxDistance);
+    should.not.exist(result.unit);
+    result.key.length.should.equal(3);
+    result.key[0].should.equal('and');
+    result.key[1].should.equal('0');
+    result.key[2].should.equal('location');
+  });
+
+  it('should handle logical operators with location and maxDistance', function() {
```

Comment:
not nit-picky. If that's the way it should be, that's the way it should be. ;) Perhaps there should be an issue opened to refactor the other tests so that others adding tests can use existing tests as an example of good naming?